src/Compilers/CSharp/Portable/Parser/QuickScanner.cs

```diff
@@ -273,7 +273,7 @@ private SyntaxToken CreateQuickToken()
         // # is marked complex as it may start directives.
         // PERF: Use byte instead of CharFlags so the compiler can use array literal initialization.
         //       The most natural type choice, Enum arrays, are not blittable due to a CLR limitation.
-        private static readonly byte[] s_charProperties = new[]
+        private static ReadOnlySpan<byte> CharProperties => new[]
```

Comment:
```suggestion
        private static ReadOnlySpan<byte> CharProperties { get; } new[]
```

The current syntax is introducing a new array allocation on ever access. That is a significant regression. Need to use the ` { get; } =` version so that it's a single allocation.

Reply:
@jaredpar This performance analyzer seems to be doing that intentionally per https://github.com/dotnet/runtime/issues/33780#issuecomment-813009032. But I don't think I understand the reasoning behind that.

Reply:
In that case though the expression itself is a `Span<T>` hence the comment is correct, you can't store it in a field. It also has no allocation (it's just loading read-only memory into a `Span<T>`). In this case the RHS is an array allocation. That can be stored as a field.

Reply:
Yep that's why I resolved my comment. My brain woke up and I remembered we'd done this optimization.